### PR TITLE
Added GAUS concept explanation 

### PR DESCRIPTION
**Pull request by clasanson**
I have added a file describing the GAUS concepts and also added a section in the developer documentation  readme.md file witch links the newly created document.

### CODE REVIEW

File: docs/concepts.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# GAUS Concepts
+
+There are a few concepts that you need to get acquainted with before setting up your first product in GAUS.
+
+Parameters and update-type are used as essential parts the product definition.
+Parameters and update-types are used to group devices into logical sets that can be operated on, e.g. make an update only for a certain set of devices.
+
+## Parameters
+
+Parameters are used to distinguish your individual devices from each other and there are only one parameter that is mandatory, Device ID. Device ID can be either something that already exists in your product like:
```
Comment (quikksilver) [inline docs/concepts.md]: Rewrite about Device Id and parameter

File: docs/concepts.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+* mac address
+* serial-number
+
+If your device does not have a unique identifier GAUS will assign one for it when the device does the registration. This parameter is mandatory when using the device api.
```
Comment (quikksilver) [inline docs/concepts.md]: GAUS will always create an unique identify called device-guid

File: docs/concepts.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+* Customization-file
+* Localization-file
+* Modem software
+* resource-file
+* applications
+* configuration-file
+
+
+## Usage of parameters and update-types
+A device that is connected to GAUS will typically periodically exercise the check-for-update api-call, except for the authentication parameters only the device id and at least one update-type is mandatory but since both parameters and update-types can be used for distribution and analysis the recommendation is to let all api calls contain the parameters and update-type versions so that you will be able to do the logical groups you want to later when distributing software updates to your devices.
```
Comment (quikksilver) [inline docs/concepts.md]: There are three "parameters" that are decorated by the system: device-id, device-guid and product-guid

Comment (clasanson) [reply docs/concepts.md]: Added a sentence about server decoration.

File: docs/concepts.md (hunk 2 above)
Comment (clasanson) [reply docs/concepts.md]: Removed text about device id to keep it simple. added a link to the register.md for further reading.

File: docs/concepts.md (hunk 1 above)
Comment (clasanson) [reply docs/concepts.md]: Rewritten

**Pull request merged**